### PR TITLE
Add get_conv_outsize and get_deconv_outsize to doc

### PR DESCRIPTION
**Pull request by mitmul**
`get_conv_outsize` and `get_deconv_outsize` is useful to check the output size before actual forward computations.
This PR adds them to the docs.

### CODE REVIEW

**Comment by niboshi** [comment]
Please fix flake8 error.

**Comment by mitmul** [comment]
@niboshi Thanks for the comment. I fixed it.

**Comment by mitmul** [comment]
@niboshi Could you merge this?

**Comment by niboshi** [comment]
* AppVeyor is failing. Please rebase.
* Please write some tests.
* A change to `.gitignore` is not relevant, isn't it?

**Comment by mitmul** [comment]
@niboshi Thanks, I fixed those two points.

File: chainer/utils/conv.py [hunk 1]
```diff
@@ -13,6 +31,23 @@ def get_conv_outsize(size, k, s, p, cover_all=False, d=1):
 
 
 def get_deconv_outsize(size, k, s, p, cover_all=False):
+    """Calculate output size of deconvolution.
```
Comment (niboshi) [inline chainer/utils/conv.py]: Calculate -> Calculates

File: chainer/utils/conv.py [hunk 2]
```diff
@@ -13,6 +31,23 @@ def get_conv_outsize(size, k, s, p, cover_all=False, d=1):
...
+
+    This functions takes the size of input feature map, kernel, stride, and
```
Comment (niboshi) [inline chainer/utils/conv.py]: functions -> function
calculate -> calculates

File: chainer/utils/conv.py [hunk 3]
```diff
@@ -13,6 +31,23 @@ def get_conv_outsize(size, k, s, p, cover_all=False, d=1):
...
+    pooling, then calculate the output feature map size.
```
Comment (niboshi) [inline chainer/utils/conv.py]: I think it's better to explicitly mention that this function only calculates for a single dimension.

File: chainer/utils/conv.py [hunk 4]
```diff
@@ -5,6 +5,24 @@
 
 
 def get_conv_outsize(size, k, s, p, cover_all=False, d=1):
+    """Calculate output size of convolution.
```
Comment (niboshi) [inline chainer/utils/conv.py]: Ditto. the same points as `get_deconv_outsize`

File: chainer/utils/conv.py [hunk 5]
```diff
@@ -13,6 +31,23 @@ def get_conv_outsize(size, k, s, p, cover_all=False, d=1):
...
+    This functions takes the size of input feature map, kernel, stride, and
+    pooling, then calculate the output feature map size.
+
+    Args:
+        size (int): The size of input feature map. It usually is the length of
+            a side of feature map.
+        k (int): The size of deconvolution kernel.
+        s (int): The size of stride.
+        p (int): The size of padding.
+        cover_all (bool): Use cover_all option or not.
```
Comment (niboshi) [inline chainer/utils/conv.py]: Please use double-backtick: ` ``cover_all`` `

File: chainer/utils/conv.py (hunk 4 above)
Comment (mitmul) [reply chainer/utils/conv.py]: Thanks, fixed the typo.

File: chainer/utils/conv.py (hunk 1 above)
Comment (mitmul) [reply chainer/utils/conv.py]: Thanks, fixed it

**Comment by niboshi** [comment]
Please consider the review.
Additionally, please add `seealso` link to each other.

File: chainer/utils/conv.py (hunk 2 above)
Comment (mitmul) [reply chainer/utils/conv.py]: 👍

File: chainer/utils/conv.py (hunk 3 above)
Comment (mitmul) [reply chainer/utils/conv.py]: What do you mean? Could you suggest some concrete sentences?

**Comment by mitmul** [comment]
@niboshi Thanks! Fixed and added seealso links.

File: chainer/utils/conv.py (hunk 3 above)
Comment (niboshi) [reply chainer/utils/conv.py]: Something like this:

"This function takes the size of input feature map, kernel, stride, and pooling of one particular dimension, then calculates the output feature map size of that dimension. "

Comment (mitmul) [reply chainer/utils/conv.py]: Thanks! I used your words.

**Comment by mitmul** [comment]
Fixed the last point.

**Comment by niboshi** [comment]
Thanks!
Please fix `get_conv_outsize` as well.

Also
``` ``cover_all`` (bool): Use cover_all option or not. ```
should be 
``` cover_all (bool): Use ``cover_all`` option or not. ```

**Comment by mitmul** [comment]
@niboshi Sorry, fixed them!

**Comment by niboshi** [comment]
LGTM!

**Comment by niboshi** [comment]
jenkins, test this please

**Comment by niboshi** [comment]
LGTM!

**Pull request merged**